rcode-reader/src/main/java/eu/rssw/pct/elements/MethodParameter.java

```diff
@@ -14,31 +14,50 @@
   private static final int PARAMETER_HANDLE = 2;
   private static final int PARAMETER_BIND = 4;
 
-  private int extent;
+  public static final int PARAMETER_INPUT = 6028;
+  public static final int PARAMETER_INOUT = 6110;
+  public static final int PARAMETER_OUTPUT = 6049;
+  public static final int PARAMETER_BUFFER = 1070;
+  
+  private final int paramNum;
+  private final int extent;
+  private final int flags;
+  private final int parameterType;
+  private final int paramMode;
+  private final int dataType;
+  private final String dataTypeName;
   // private int crc;
-  private int flags;
-  private int parameterType;
-  private int paramMode;
-  private int dataType;
   // private int fullNameLength;
-  private int paramNum;
-
-  private String argumentName;
-
-  protected MethodParameter(byte[] segment, int currentPos, int textAreaOffset, ByteOrder order) {
-    this.parameterType = ByteBuffer.wrap(segment, currentPos, Short.BYTES).order(order).getShort();
-    this.paramMode = ByteBuffer.wrap(segment, currentPos + 2, Short.BYTES).order(order).getShort();
-    this.extent = ByteBuffer.wrap(segment, currentPos + 4, Short.BYTES).order(order).getShort();
-    this.dataType = ByteBuffer.wrap(segment, currentPos + 6, Short.BYTES).order(order).getShort();
-    // this.crc = ByteBuffer.wrap(segment, currentPos + 8, Short.BYTES).order(order).getShort();
-    this.flags = ByteBuffer.wrap(segment, currentPos + 10, Short.BYTES).order(order).getShort();
-    // this.fullNameLength = ByteBuffer.wrap(segment, currentPos + 12, Short.BYTES).order(order).getShort();
+
+  public MethodParameter(int num, String name, int type, int mode, int flags, int dataType, String dataTypeName,
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Constructor has 8 parameters, which is greater than 7 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS00107)